app/server/utils/ussd/directory_listing_processor.py

```diff
@@ -1,39 +1,35 @@
 import threading
+from sqlalchemy import text
 
-from server import message_processor
+from server import message_processor, db
 from server.utils.i18n import i18n_for
 from server.models.user import User
-from server.models.custom_attribute_user_storage import CustomAttributeUserStorage
 from server.models.transfer_usage import TransferUsage
 from server.models.transfer_account import TransferAccount
+from server.utils.user import default_token
 
 
 def user_directory_listing(user):
-    # TODO: add custom attribute bio
-    return f"{user.phone}"
+    bio = user.custom_attributes.filter_by(name='bio').first()
+    if bio is None:
+        return user.phone
+    else:
+        return f"{user.phone} {bio.value}"
 
 
-class DirectoryListingProcessor(threading.Thread):
+class DirectoryListingProcessor(object):
     def __init__(self, user: User, transfer_usage: TransferUsage):
         self.recipient = user
         self.selected_business_category = transfer_usage
 
-        threading.Thread.__init__(self)
```

Comment:
got rid of it being on a background thread, otherwise i get these fun exceptions:

```
Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.6.5_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/Users/owen/workspace/sempo/SempoBlockchain/app/server/utils/ussd/directory_listing_processor.py", line 30, in run
    self.runnable()
  File "/Users/owen/workspace/sempo/SempoBlockchain/app/server/utils/ussd/directory_listing_processor.py", line 82, in send_directory_listing
    transfer_account = TransferAccount.query.get(self.recipient.default_transfer_account_id)
  File "/Users/owen/workspace/sempo/SempoBlockchain/venv/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", line 519, in __get__
    return type.query_class(mapper, session=self.sa.session())
  File "/Users/owen/workspace/sempo/SempoBlockchain/venv/lib/python3.6/site-packages/sqlalchemy/orm/scoping.py", line 78, in __call__
    return self.registry()
  File "/Users/owen/workspace/sempo/SempoBlockchain/venv/lib/python3.6/site-packages/sqlalchemy/util/_collections.py", line 1012, in __call__
    return self.registry.setdefault(key, self.createfunc())
  File "/Users/owen/workspace/sempo/SempoBlockchain/venv/lib/python3.6/site-packages/sqlalchemy/orm/session.py", line 3214, in __call__
    return self.class_(**local_kw)
  File "/Users/owen/workspace/sempo/SempoBlockchain/venv/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", line 136, in __init__
    self.app = app = db.get_app()
  File "/Users/owen/workspace/sempo/SempoBlockchain/venv/lib/python3.6/site-packages/flask_sqlalchemy/__init__.py", line 982, in get_app
    'No application found. Either work inside a view function or push'
RuntimeError: No application found. Either work inside a view function or push an application context. See http://flask-sqlalchemy.pocoo.org/contexts/.
```

what should the proper fix be?

Reply:
Might be fixed by the `copy_current_request_context` decorator:
https://flask.palletsprojects.com/en/1.1.x/api/#flask.copy_current_request_context

Reply:
do we want this task to run in a different thread since it's doing a massive query?

Reply:
i'll leave it as sync for now since everything else is also sync. if we find it to be too slow we can throw it in a different thread again